lib/bcu/command/upgrade.rb

```diff
@@ -56,7 +56,15 @@ def run_process(_args, options)
       # In interactive flow we're not sure if we need to clean up
       cleanup_necessary = !options.interactive
 
+      batched = []
+
       outdated.each do |app|
+        upgrade app, options, state_info, batched
+      end
+
+      # upgrade deferred applications
+      options.interactive = false
```

Comment:
I would also suggest (just to be sure), keeping the original value in memory and then restore the original value after the upgrade happened. That way if there is anything after the code is executed depending on that flag, it will behave the same since the user requested that.